mongo/mongo_test.go

```diff
@@ -72,52 +72,45 @@ func TestMongoHelpers(t *testing.T) {
 	t.Run("transform and ensure ID", func(t *testing.T) {
 		t.Run("newly added _id should be first element", func(t *testing.T) {
 			doc := bson.D{{"foo", "bar"}, {"baz", "qux"}, {"hello", "world"}}
-			want := bsonx.Doc{
-				{"_id", bsonx.Null()}, {"foo", bsonx.String("bar")},
-				{"baz", bsonx.String("qux")}, {"hello", bsonx.String("world")},
-			}
 			got, id, err := transformAndEnsureID(bson.DefaultRegistry, doc)
 			assert.Nil(t, err, "transformAndEnsureID error: %v", err)
 			oid, ok := id.(primitive.ObjectID)
 			assert.True(t, ok, "expected returned id type %T, got %T", primitive.ObjectID{}, id)
-			want[0] = bsonx.Elem{"_id", bsonx.ObjectID(oid)}
-			assert.Equal(t, got, want, "expected document %v, got %v", got, want)
-		})
-		t.Run("existing _id should be first element", func(t *testing.T) {
-			doc := bson.D{{"foo", "bar"}, {"baz", "qux"}, {"_id", 3.14159}, {"hello", "world"}}
-			want := bsonx.Doc{
-				{"_id", bsonx.Double(3.14159)}, {"foo", bsonx.String("bar")},
-				{"baz", bsonx.String("qux")}, {"hello", bsonx.String("world")},
+			wantDoc := bson.D{
+				{"_id", oid}, {"foo", "bar"},
+				{"baz", "qux"}, {"hello", "world"},
 			}
-			got, id, err := transformAndEnsureID(bson.DefaultRegistry, doc)
-			assert.Nil(t, err, "transformAndEnsureID error: %v", err)
-			_, ok := id.(float64)
-			assert.True(t, ok, "expected returned id type %T, got %T", float64(0), id)
-			assert.Equal(t, got, want, "expected document %v, got %v", got, want)
+			_, wantBSON, _ := bson.MarshalValue(wantDoc)
```

Comment:
Here and in the tests for `transformAggregatePipeline`, I `MarshalValue(...)` instead of just writing the document out in `bsoncore` format. I could do the latter, but this seems more readable to me and is not code that will be run by users/needs to be performant.

Reply:
Agreed, readability is more important than performance in tests. I do have a concern that ignoring the returned `error` might result in unexpected test conditions in the case that marshaling fails. Consider capturing the returned `error` and assert it's `nil`.

Reply:
Added non-nil assertions here and in `transformAggregatePipeline`.